### DISCUSSION TITLE
Git focused history?

### DISCUSSION THREAD

**Discussion by mitchellmoss** [cline/cline#520] [Feature Requests]
Is there a way to have an auto-commit behavoir like Aider? I like the timeline based history but if the claude goes off on a tangent I would be nice to reset all files to the same state.

**Comment by TomLucidor** [comment]
Seconding this to be mixed with TDD and "Architecture Mode" as a complete development suite

**Comment by dcbartlett** [comment]
Checkpoints have been added.  This is backed by Git and when a checkpoint is selected to revert back to, Cline uses the Git repo for the checkpoints to revert your project back to the state for that checkpoint.
